modules/types_proto/src/numeric_value.cpp

```diff
@@ -0,0 +1,91 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include "hephaestus/types_proto/numeric_value.h"  //NOLINT(misc-include-cleaner)
+
+#include <cstdint>
+
+#include "hephaestus/types/proto/numeric_value.pb.h"
+#include "hephaestus/utils/exception.h"
+
+namespace heph::types::proto {
+
+template <>
+auto toProto(NumericValue& proto_value, int32_t value) -> void {
```

Comment:
not sure if this is the best way to solve this. Wouln't it make more sense to implement a concept that holds true for all types where the proto type has a `.set_value` function? same with `fromProto`?

Reply:
I'm torn here. In general, agreed with your statement. 

In this particular case, bounds only make sense for numerical values. Given that this statement is true for other functionality as well, which will be templated on numeric types only, a complete numeric type serialization seems useful.

@filippobrizzi @floriantschopp please wager in here if we want a complete numeric types serialization along the lines proposed here.

Reply:
but for different types there are different functions name, so you cannot really generalize it the only other possible approach is to use `if constexpr` but I think that is less readable

Reply:
We actually need to specialize somewhere, so the current approach is as good as specializing elsewhere. We keep it for now.